kafka/src/main/java/org/apache/apex/malhar/kafka/KafkaSinglePortExactlyOnceOutputOperator.java

```diff
@@ -182,6 +182,10 @@ public void teardown()
   @Override
   public void endWindow()
   {
+    if (windowId <= windowDataManager.getLargestCompletedWindow()) {
+      return;
+    }
+
```

Comment:
The check for "windowId > windowDataManager.getLargestCompletedWindow()" in the next line is now redundant.